File: patterns/trennung-von-verhaltensweisen/rules-js/trennung-von-verhaltensweisen-cloudy.js
rules.JSRule({
  name: "Is it cloudy outside?",
  triggers: [triggers.ItemStateChangeTrigger("vCloudiness")],
  execute: (event) => {
    const newState = parseFloat(items.getItem("vCloudiness").state) > 50 ? "ON" : "OFF";
    if (newState !== items.getItem("vIsCloudy").state) {
      items.getItem("vIsCloudy").postUpdate(newState);
    }
  }
});

File: patterns/trennung-von-verhaltensweisen/rules-python/trennung_von_verhaltensweisen_cloudy.py
from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger


@rule(triggers=[ItemStateChangeTrigger("vCloudiness")])
class IsItCloudyOutside:
    def execute(self, module, input):
        cloudiness = float(str(Registry.getItem("vCloudiness").state))
        new_state = "ON" if cloudiness > 50 else "OFF"

        is_cloudy = Registry.getItem("vIsCloudy")
        if new_state != str(is_cloudy.state):
            is_cloudy.postUpdate(new_state)
